src/dash/models/DashManifestModel.js

```diff
@@ -292,6 +292,24 @@ function DashManifestModel(config) {
         return adaptation[DashConstants.CENC_DEFAULT_KID];
     }
 
+    function getLabelForAdaptation(adaptation) {
+        if (adaptation.lang && !adaptation.Label_asArray) {
+            return adaptation.lang;
+        }
+
+        if (adaptation.lang && adaptation.Label_asArray) {
+            const foundMatchingLabel = adaptation.Label_asArray.find((label) => label.lang === adaptation.lang);
```

Comment:
`Array.find` isn't available on IE11 :sob:

Also, I think it would be useful to handle the case of a single `Label` with no `lang` information. Perhaps something like this?
```suggestion
            if (adaptation.Label_asArray.length === 1) {
                return adaptation.Label_asArray[0].__text;
            }
            for (let i = 0; i < adaptation.Label_asArray.length; i++) {
                if (adaptation.Label_asArray[i].lang === adaptation.lang) {
                    return adaptation.Label_asArray[i].__text;
                }
            }
```

or if you're feeling fancy, maybe a `reduce`
```suggestion
            return adaptation.Label_asArray.reduce((acc, cur) => {
                return acc || curr.lang === adaptation.lang ? curr.__text : null;
            }, null) || adaptation.Label_asArray[0].__text;
```

Reply:
@squarebracket - I thought we had polyfills or babel - but looks like not.  Updated!